cmd/destroy.go

```diff
@@ -60,14 +76,66 @@ func runDestroyCmd(cmd *cobra.Command, args []string) {
 	checkErr(validateGroupSelectionFlags(bp), ctx)
 	checkErr(shell.ValidateDeploymentDirectory(bp.Groups, deplRoot), ctx)
 
+	destroyRunner(deplRoot, artifactsDir, bp, ctx)
+}
+
+func destroyRunner(deplRoot string, artifactsDir string, bp config.Blueprint, ctx *config.YamlCtx) {
+	maxRetries := 1
+	if robustDestroy {
+		maxRetries = 3
+	}
+
+	for attempt := 1; attempt <= maxRetries; attempt++ {
+		logging.Info("Destroy attempt %d of %d", attempt, maxRetries)
+
+		destroyFailed, packerManifests := destroyGroupsFunc(deplRoot, artifactsDir, bp, ctx)
+
+		if !destroyFailed {
+			logging.Info("Successfully destroyed all selected groups.")
+			modulewriter.WritePackerDestroyInstructions(os.Stdout, packerManifests)
+			return // Exit runDestroyCmd successfully
+		}
+
+		if attempt == maxRetries {
+			logging.Fatal("Destruction of %q failed after %d attempts", deplRoot, maxRetries)
+		}
+		logging.Info("Retrying destroy...")
+	}
+}
+
+func groupHasNetworkModule(group config.Group) bool {
+	for _, module := range group.Modules {
+		if strings.HasPrefix(module.Source, "modules/network/") || strings.HasPrefix(module.Source, "community/modules/network/") {
+			return true
+		}
+	}
+	return false
+}
+
+func destroyGroups(deplRoot string, artifactsDir string, bp config.Blueprint, ctx *config.YamlCtx) (bool, []string) {
 	// destroy in reverse order of creation!
 	packerManifests := []string{}
+	destroyFailed := false
 	for i := len(bp.Groups) - 1; i >= 0; i-- {
 		group := bp.Groups[i]
 		if !isGroupSelected(group.Name) {
 			logging.Info("skipping group %q", group.Name)
 			continue
 		}
+
+		if robustDestroy && groupHasNetworkModule(group) {
+			projectID, deploymentName, err := getProjectAndDeploymentVars(bp.Vars)
+			if err != nil {
+				logging.Error("Skipping firewall cleanup: could not get required variables. %v", err)
```

Comment:
The log statement is not right as the the step is not skipped but retried again by setting the failed flag